hardware_interface/include/hardware_interface/component_parser.hpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__COMPONENT_PARSER_HPP_
+#define HARDWARE_INTERFACE__COMPONENT_PARSER_HPP_
+
+#include <tinyxml2.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
```

Comment:
This has to be like this otherwise cpplint is unhappy with message: "Found C system header after C++ system header. Should be: component_parser.h, c system, c++ system, other."